### PR TITLE
Fix `get_user_description`

### PR DESCRIPTION
**Pull request by zmeir**
## Type
bug_fix

___

## Description
- Fixed the `get_user_description` method in `git_provider.py` to handle changes in the headers from "PR Type"/"PR Description"/etc to "Type"/"Description"/etc.
- Introduced a new helper method `_is_generated_by_pr_agent` to check if the description was generated by the pr-agent.
- Updated the extraction of the user description to match the new header format.
- Updated the `_prepare_pr_answer` method in `pr_description.py` to change the formatting of the key_publish variable. The `capitalize` method was replaced with the `title` method to ensure each word in the key is capitalized.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>git_provider.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        pr_agent/git_providers/git_provider.py<br><br>

**The method `get_user_description` was updated to handle <br>changes in the headers from "PR Type"/"PR Description"/etc <br>to "Type"/"Description"/etc. A new helper method <br>`_is_generated_by_pr_agent` was introduced to check if the <br>description was generated by the pr-agent. The extraction of <br>the user description was also updated to match the new <br>header format.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/561/files#diff-52d45f12b836f77ed1aef86e972e65404634ea4e2a6083fb71a9b0f9bb9e062f"> +7/-3</a></td>

</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>pr_description.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        pr_agent/tools/pr_description.py<br><br>

**The method `_prepare_pr_answer` was updated to change the <br>formatting of the key_publish variable. The `capitalize` <br>method was replaced with the `title` method to ensure each <br>word in the key is capitalized.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/561/files#diff-66130451adce278a098a5770e2db52b12050205fcd4d80c13f813615d4449abb"> +1/-1</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## User description
The headers changed from "PR Type"/"PR Description"/etc to "Type"/"Description"/etc

### CODE REVIEW

File: pr_agent/tools/pr_description.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def _prepare_pr_answer(self) -> Tuple[str, str]:
                 value = self.file_label_dict
                 key_publish = "PR changes walkthrough"
             else:
-                key_publish = key.rstrip(':').replace("_", " ").capitalize()
+                key_publish = key.rstrip(':').replace("_", " ").title()
```
Comment (mrT23) [inline pr_agent/tools/pr_description.py:300]: i don't think this is good. 
i do want to capitalize only the first letter

Comment (mrT23) [reply pr_agent/tools/pr_description.py:300]: ("PR changes walkthrough"
instead of
"PR Changes Walkthrough")

Comment (zmeir) [reply pr_agent/tools/pr_description.py:300]: Sure I will revert it, I just though since those are section titles it will be more fitting.

**Comment by zmeir** [comment]
@mrT23 thanks for the review - I reverted the use of `title()` to `capitalize()`, and also made the logic to find the user description case-insensitive, so that if you change the caseness of the headers in the future it will still work (e.g. if you decide to change `## Description` to `## description`, `## PR Description` or `## PR description`, etc.).

**Review by mrT23 — APPROVED** [review]

**Pull request merged**